pkg/conditions/poll_test.go

```diff
@@ -0,0 +1,76 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package conditions_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/talos-systems/talos/pkg/conditions"
+)
+
+func TestPollingCondition(t *testing.T) {
+	t.Parallel()
+
+	t.Run("OK", func(t *testing.T) {
```

Comment:
Also, how do you feel about `testing` sub-tests and `testify/suite`? Do we want to replace the latter with the former? Previously I had a lot of problems with a combination of parallel tests + logging in tests + suites, a moving to sub-tests solved them completely. I'm suspicious about `testify/suite` since then :)

Reply:
I think I used sub-tests in all kinds of ways, including inside `testify/suite` and launching suite multiple times as sub-tests, and it all worked (so far), but I might not have hit the issue yet.